boefjes/plugins/kat_webpage_analysis/main.py

```diff
@@ -33,7 +32,7 @@ def run(boefje_meta: BoefjeMeta) -> List[Tuple[set, Union[bytes, str]]]:
     else:
         # Fall back to old hack-ip-into-url behavior, for either https with no adapter, or http.
         if ip:
-            url_parts = url_parts._replace(netloc=ip)
+            url_parts = url_parts._replace(netloc="[%s]" % ip)
```

Comment:
As far as I know [] only work with IPv6, not with IPv4? At least wget won't accept it:

```
% wget 'http://[127.0.0.1]/'
http://[127.0.0.1]/: Invalid IPv6 numeric address.
```

So we probably need to check if the address is IPv4 or IPv6.

Reply:
Requests has no problem with the notation with brackets for both ipv6 and ipv4.

Reply:
> Requests has no problem with the notation with brackets for both ipv6 and ipv4.

Are you sure about this? I tried to test this using requests 2.28.1 but I got a parse error for the bracketed IPv4 address but not for IPv6

Reply:
Seems to work in KAT